### ISSUE TITLE
nunit3-console runner displays modal dialog breaking CI automation

### ISSUE THREAD

**Issue by githubuser185** [nunit/nunit-console#414]
My company is using the nunit 3.8 console runner to execute production unit tests. We use the console runner to run the tests in an automated way uder the control of a CI server, where the unit tests continue to cycle with no human interaction.

On a suite of 5000 tests that are defined in a single unit test DLL, we are finding that the nunit3-console runner is for some reason displaying a modal dialog which blocks test execution until a CI server timeout occurs to kill the runner process.  

The command line we are using on the CI server is:
"C:\Program Files (x86)\NUnit.org\nunit-console\nunit3-console.exe" "C:\development\main\Bin\NextGen.Interfaces.UnitTests.dll" --x86 --config=Release --framework=4.0.30319 --result="C:\buildlogs\InterfaceMainDev-nunit-results.xml;format=nunit2" --work="C:\development\main\Bin" --dispose-runners --timeout=120000

When the modal dialog appears, I used Spy++ as a tool to find which process owns the modal dialog and have found the "NUNIT-AGENT-X86" process is the dialog window owner, where the dialog includes an "OK" button to ack the dialog to allow the testing to continue.  

Has this nunit3 modal dialog problem been reported before, and if not, can someone assist on how to prevent the dialog from appearing and pausing the automated testing?
Thanks.

**Comment by mikkelbu** [comment]
There are no calls to dialogs or winforms in the console (see e.g. https://github.com/nunit/nunit-console/search?utf8=%E2%9C%93&q=dialog&type= or https://github.com/nunit/nunit-console/search?utf8=%E2%9C%93&q=winforms&type=Issues). My guess is that one of the tests trieds to show a dialog. If you run the tests with `--labels=all` you should be able to see which is the last test to be run.

**Comment by githubuser185** [comment]
Thanks for the assist, and makes sense that the problem is most likely with the unit test being executed.  I have just added the extra --labels=all arg to the automation config and will now wait for a repeat.  Issue is hard to reproduce so it might be a while before I can reply back with what I find.  It is good to know that at least the agent application does not call any functions/methods to display a dialog.

**Comment by ChrisMaddock** [comment]
Closing this, as it looks like the problem may lie outside of NUnit.

Happy to reopen if anyone has more to add. 🙂

**Comment by githubuser185** [comment]
Just thought I would reply back to let this thread know that we found one of the production DLLs being tested by a unit test DLL had an execution path that calls the .NET MessageBox as a modal dialog.   This was difficult to find since the execution path was not always being executed due to the data being used for the test.  This issue is no longer happening once we prevented the use of the MessageBox call.
